lib/ae_mdw/db/mutations/contract_call_mutation.ex

```diff
@@ -30,13 +43,45 @@ defmodule AeMdw.Db.ContractCallMutation do
 
   @spec mutate(t()) :: :ok
   def mutate(%__MODULE__{
+        contract_pk: contract_pk,
+        caller_pk: caller_pk,
         create_txi: create_txi,
         txi: txi,
         fun_arg_res: fun_arg_res,
         call_rec: call_rec
       }) do
     DBContract.call_write(create_txi, txi, fun_arg_res)
     DBContract.logs_write(create_txi, txi, call_rec)
+
+    if Contract.is_aex9?(contract_pk) and Contract.is_successful_call?(fun_arg_res) and
+         not Contract.is_non_stateful_aex9_function?(fun_arg_res.function) do
+      update_aex9_presence(
+        contract_pk,
+        caller_pk,
+        txi,
+        fun_arg_res.function,
+        fun_arg_res.arguments
+      )
+    end
```

Comment:
Perhaps something like:
```elixir
with true <- Contract.is_aex9(contract_pk),
  {:ok, fname, args} <- Contract.extract_non_stateful_function(fun_arg_res) do
  
  update_aex9_presence(...)
end
```

Dialyzer is always going to complain about these cases (and any decent statically typed language will do too), so I would try to avoid getting into a point where we would need to add ignores for every one of them

Reply:
I would avoid saying any, always, never, none in our profession (: it's not a NP-hard problem.

Good naming solution, thanks.

Reply:
Please note it won't cover semantically these cases but we need to move on so I am removing the dialyzer_ignore. 
```
def extract_non_stateful_aex9_function(%{function: _fname, result: %{error: _error}}), do: :not_found
def extract_non_stateful_aex9_function(%{function: _fname, result: %{abort: _error}}), do: :not_found
```

Reply:
> I would avoid saying any, always, never, none in our profession (: it's not a NP-hard problem.
> 
> Good naming solution, thanks.

"always", "never", "none", "all" are _**simplifier**_ words. "sometimes", "if", "when", "some" are _**complecting**_ words. Our job is to find the simplest solution (the hardest thing to do), and avoid circumventing the problem so that others have to pay for it in the future

Reply:
"any decent statically typed language will do too" is not a strong statement? ok :+1: 
next problem with any compiler I will call you (: